pennylane/ops/op_math/prod.py

```diff
@@ -68,6 +69,12 @@ def prod(*ops, do_queue=True, id=None):
     array([[ 0, -1],
            [ 1,  0]])
     """
+    if not lazy:
+        ops_simp = tuple()
```

Comment:
Since `ops_simp` is going to be dynamically updated a lot, lists are going to be more efficient than tuples.  Every time we modify the tuple, we would have to create a new object.  If we perform this with a list, python just modifies the existing object.

I don't know if the performance difference will show up for a function like this in practice, but better safe than sorry.  While lists are often overused and abused, this is a case we want mutability.

This can easily be checked running `%%timeit` on both
```
a = tuple()
for i in range(100):
    a += (i,)
```
and 
```
a = []
for i in range(100):
    a += [i]
```

Reply:
Hi Christina, I've replaced tuples by lists, please check.